designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,209 @@
+* If the lambda function is associated with a CodePipeline, the user is given an alternate prompt (see Confirmation Prompt below). If the user agrees to proceed with function deletion, they do not get an additional prompt. The function is deleted.
+
+### Deleting a CloudFormation Stack
+
+Users delete a CloudFormation Stack through the AWS Explorer by right-clicking on the Stack and selecting `Delete CloudFormation Stack`.
+
+Currently:
+* a notification message pops up with the message "Are you sure you want to delete '`STACK_NAME_HERE`'?"
+* the notification message has `Yes` and `No` buttons
+
+Proposed:
+* If the CloudFormation Stack is not associated with a CodePipeline, the same experience takes place.
+* If the CloudFormation Stack is associated with a CodePipeline, the user is given an alternate prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with Stack deletion, they do not get an additional prompt. The Stack is deleted.
+
+### Deploying a SAM Application to an existing CloudFormation Stack
+
+Users deploy a SAM Application using a Command Palette action. Through a Wizard, they are asked for details about what to deploy, and where to deploy it, including specifying a CloudFormation Stack. The stack specified can be new or may already exist. At this time, users must type the stack name in manually. There is a backlog task to replace this with a list that the user can choose a stack from.
+
+Proposed:
+* After the wizard collects all information from the user, we add a final step in the wizard that confirms all of the wizard settings before proceeding.
+* Prior to the wizard settings confirmation we put a busy indicator on the UI (and inform the user that we are performing some validations), and check to see if the Stack belongs to a CodePipeline.
+  * If it does not, the wizard settings confirmation comes up
+  * If it does, the user is given a confirmation prompt (see [Confirmation Prompt](#confirmation-prompt) below).
+    * If the user agrees to proceed with the deploy, it proceeds normally.
+    * If the user elects to back out, the wizard moves back to the step where the user can specify a different CloudFormation Stack.
+
+This approach was chosen for the following reasons:
+* it provides a place to put additional deploy wizard validations
+* performing validations in between each wizard step could introduce latency, causing an unsatisfactory experience
+
+### <a id="confirmation-prompt"></a>Confirmation Prompt
+
+The confirmation prompt contains the following messaging. See [Success criteria](#success-criteria) for the required information.
+
+---
+
+{ AWS Lambda Function | AWS CloudFormation Stack } '`friendly_name_here`' is part of an AWS CodePipeline pipeline (`associated_pipeline`).
+
+Manual modification of resources outside of an AWS CodePipeline will skip any verification steps defined in the pipeline.
```

Comment:
For deployments, maybe make a mention that a `git push` should deploy the change?

Reply:
The current messaging is consistent with the Jetbrains messaging (which went through a review pass). I think we should start with that, but I like the idea of adding suitable guidance.

Reply:
This point came up in Jetbrains messaging review as well. _If a CodeCommit repo (part of CodePipeline) is checked out locally in IDE and can be identified, we should mention that a `git push` should deploy the change._ At least, we should have a task to investigate this. I've added this point in Jetbrains toolkit's issue as well.